### PR TITLE
Portals in generator

### PR DESCRIPTION
**Pull request by jorisBarkema**
maze generator now adds portals to the maze.

### CODE REVIEW

**Comment by jorisBarkema** [comment]
I tested it and it works great I think, not 100% sure if the increased exit score works fully as intended because to test that I'd have to test several levels and see if I need to use portals.

One thing I'm struggling with is right now I think the second portal can get placed over a key tile, which would cause issues. The chances of that happening are slim, but it can't be allowed to happen at all.

**Comment by jensj12** [comment]
It looks like it'll skip placing a key if it has placed a portal. Always place the second portal on a walltile, that is, not connected to any previously placed path and continue from there like it is a normal pathway.

**Comment by jorisBarkema** [comment]
Yes I know about the skipping, I also made the doors dependent on how many keys had ben placed instead of numKeys, so that one woulnd't cause any insolvable mazes just a very small chance of having less keys. I'll try to find a more elegant solution for that too though.

As for placing it on walls, I tried adding a while (secondporatl != walltile) kind of  loop in the determinesecondportalposition method, but that would cause the whole generator to get stuck. Even if I let it check against just keytiles, it would get stuck there.

**Comment by jorisBarkema** [comment]
I started working on what you said, but I noticed the generator now makes impossible mazes noticably more often than before, I think 1/4 times or something like that. Don't think that's acceptable, but I don't have a clue on how to fix that.

**Comment by jensj12** [comment]
An extra check in AddLoopTile should fix that, check if there's a wall in between to prevent blocking an existing path.

**Comment by jorisBarkema** [comment]
Oooohh yes that could be it, I'll check now.
edit: Nope, that check's there already in MazeGenMain() before addlooptile is called.

File: MeesGame/Gameplay/Generator/MazeGeneratorMain.cs [hunk 1]
```diff
@@ -130,5 +97,116 @@ private void MazeGenMain()
...
+                    tileToAdd = new GuardTile();
+            }
+            tiles.Add(tileToAdd, PointBetween(one, two).X, PointBetween(one, two).Y);
+
+            // If a loop is added, the tiles are only behind the doors which both of them are behind.
+            // This should continue for further along the hallways of both loops, but doign that would require some form of flood-fill 
+            // which I don't have time for.
+            if(tileToAdd.TileType == TileType.Floor)
+            {
+                behindDoorInfo[one.X, one.Y] &= behindDoorInfo[two.X, two.Y];
```
Comment (jensj12) [inline MeesGame/Gameplay/Generator/MazeGeneratorMain.cs]: Floortiles are only chosen if the difference is 0, so this isn't needed. I have removed it.

**Review by jorisBarkema — COMMENTED** [review]
Looks good

**Review by jensj12 — APPROVED** [review]

**Pull request merged**